storage/Devices/LuksImpl.cc

```diff
@@ -603,6 +617,14 @@ namespace storage
 		ST_THROW(Exception("invalid encryption type"));
 	}
 
+	if (get_cipher() == "aegis128-random" || !get_integrity().empty()) {
+	    if (get_cipher() != "aegis128-random")
+		ST_THROW(Exception("when integrity is set it is expected aegis128-random cipher"));
+
+	    if (get_integrity() != "aead")
+		ST_THROW(Exception("for now only AEAD integrity is supported"));
+	}
```

Comment:
Checking this during commit is very late. Likely a better place is in check().

And please add a document explaining the situation (what works, what does not work, what kernel is needed, ...) in the doc directory.